demo/node/node_inprocess.go

```diff
@@ -0,0 +1,288 @@
+		drand, err := core.NewDrand(fs, conf)
+		if err != nil {
+			return err
+		}
+		l.daemon = drand
+	} else {
+		drand, err := core.LoadDrand(fs, conf)
+		if err != nil {
+			return err
+		}
+		drand.StartBeacon(true)
+		l.daemon = drand
+	}
+	return nil
+}
+
+func (l *LocalNode) PrivateAddr() string {
+	return l.privAddr
+}
+
+func (l *LocalNode) PublicAddr() string {
+	return l.pubAddr
+}
+
+func (l *LocalNode) Index() int {
+	return l.i
+}
+
+func (l *LocalNode) RunDKG(nodes, thr int, timeout string, leader bool, leaderAddr string, beaconOffset int) *key.Group {
+	cl, err := net.NewControlClient(l.ctrlAddr)
+	if err != nil {
+		l.log.Error("drand", "can't instantiate control client", "err", err)
+		return nil
+	}
+
+	p, _ := time.ParseDuration(l.period)
+	t, _ := time.ParseDuration(timeout)
+	var grp *drand.GroupPacket
+	if leader {
+		grp, err = cl.InitDKGLeader(nodes, thr, p, t, nil, secretDKG, beaconOffset)
+	} else {
+		leader := net.CreatePeer(leaderAddr, l.tls)
+		grp, err = cl.InitDKG(leader, nodes, thr, t, nil, secretDKG)
+	}
+	if err != nil {
+		l.log.Error("drand", "dkg run failed", "err", err)
+		return nil
+	}
+	kg, _ := key.GroupFromProto(grp)
+	return kg
+}
+
+func (l *LocalNode) GetGroup() *key.Group {
+	cl, err := net.NewControlClient(l.ctrlAddr)
+	if err != nil {
+		l.log.Error("drand", "can't instantiate control client", "err", err)
+		return nil
+	}
+	grp, err := cl.GroupFile()
+	if err != nil {
+		l.log.Error("drand", "can't  get group", "err", err)
+		return nil
+	}
+	group, err := key.GroupFromProto(grp)
+	if err != nil {
+		l.log.Error("drand", "can't deserialize group", "err", err)
+		return nil
+	}
+	return group
+}
+
+func (l *LocalNode) RunReshare(nodes, thr int, oldGroup string, timeout string, leader bool, leaderAddr string, beaconOffset int) *key.Group {
+	cl, err := net.NewControlClient(l.ctrlAddr)
+	if err != nil {
+		l.log.Error("drand", "can't instantiate control client", "err", err)
+		return nil
+	}
+
+	t, _ := time.ParseDuration(timeout)
+	var grp *drand.GroupPacket
+	if leader {
+		grp, err = cl.InitReshareLeader(nodes, thr, t, secretReshare, oldGroup, beaconOffset)
+	} else {
+		leader := net.CreatePeer(leaderAddr, l.tls)
+		grp, err = cl.InitReshare(leader, nodes, thr, t, secretReshare, oldGroup)
+	}
+	if err != nil {
+		l.log.Error("drand", "reshare failed", "err", err)
+		return nil
+	}
+	kg, _ := key.GroupFromProto(grp)
+	return kg
+}
+
+func (l *LocalNode) GetCokey(group string) bool {
+	cl, err := net.NewControlClient(l.ctrlAddr)
+	if err != nil {
+		l.log.Error("drand", "can't instantiate control client", "err", err)
+		return false
+	}
+	key, err := cl.CollectiveKey()
+	if err != nil {
+		l.log.Error("drand", "can't get cokey", "err", err)
+		return false
+	}
+	sdist := hex.EncodeToString(key.GetCoKey())
+	fmt.Printf("\t- Node %s has cokey %s\n", l.PrivateAddr(), sdist[10:14])
+	return true
+}
+
+func (l *LocalNode) Ping() bool {
+	cl, err := net.NewControlClient(l.ctrlAddr)
```

Comment:
By creating the control client once (so the connection is performed once) we might get some performance benefit _on circleCI_ :)